azurelinuxagent/common/cgroup.py

```diff
@@ -13,104 +13,139 @@
 # limitations under the License.
 #
 # Requires Python 2.6+ and Openssl 1.0+
-
+import os
 import re
 
+from azurelinuxagent.common import logger
 from azurelinuxagent.common.exception import CGroupsException
+from azurelinuxagent.common.future import ustr
 from azurelinuxagent.common.osutil import get_osutil
+from azurelinuxagent.common.utils import fileutil
 
-re_user_system_times = re.compile('user (\d+)\nsystem (\d+)\n')
+re_user_system_times = re.compile(r'user (\d+)\nsystem (\d+)\n')
 
 # The metric classes (Cpu, Memory, etc) can all assume that CGroups is enabled, as the CGroupTelemetry
 # class is very careful not to call them if CGroups isn't enabled. Any tests should be disabled if the osutil
 # is_cgroups_support() method returns false.
 
 
 class CGroup(object):
-    def __init__(self):
-        pass
-
-    def _get_cgroup_file(self, hierarchy, file_name):
-        return os.path.join(self.cgroups[hierarchy], file_name)
+    def __init__(self, extension_name, cgroup_path, controller_type):
+        self.name = extension_name
```

Comment:
what is the name used for? i think i saw it only on an error message that also includes the path... the path should be enough to identify the cgroup.

If we do need the name it should not be "extension_name", since we will also have cgroups for the agent, services, etc

Reply:
We need the name in the Cgroup metrics. Yes, I'll change the parameter of the `__init__` to `name` rather than `extension_name`